src/components/Menus/Sidebar/DownloadPanel/DownloadPanel.test.js

```diff
@@ -0,0 +1,84 @@
+import DownloadPanel from "./DownloadPanel";
+import { fireEvent, render, waitFor, screen } from "@testing-library/react";
+import { Provider } from "react-redux";
+import { MemoryRouter } from "react-router";
+import configureStore from "redux-mock-store";
+import FileSaver from "file-saver";
+
+jest.mock("file-saver");
+jest.mock("jszip");
+jest.mock("jszip-utils", () => ({
+  getBinaryContent: jest.fn(),
+}));
+
+let container;
+describe("DownloadPanel", () => {
+  beforeEach(() => {
+    const middlewares = [];
+    const mockStore = configureStore(middlewares);
+    const initialState = {
+      editor: {
+        project: {
+          name: "My epic project",
+          identifier: "hello-world-project",
+          components: [
+            {
+              name: "main",
+              extension: "py",
+              content: "print('hello world')",
+            },
+          ],
+          image_list: [
+            {
+              url: "a.com/b",
+            },
+          ],
+        },
+      },
+    };
+    const store = mockStore(initialState);
+
+    container = render(
+      <Provider store={store}>
+        <MemoryRouter>
+          <DownloadPanel />
+        </MemoryRouter>
+      </Provider>,
+    );
+  });
+  test("Renders the correct heading", () => {
+    expect(container.getByText("downloadPanel.heading")).not.toBeNull();
+  });
+
+  test("Renders the correct subtitle", () => {
+    expect(container.getByText("downloadPanel.subtitle")).not.toBeNull();
+  });
+
+  test("Renders the log-in hint", () => {
+    expect(container.getByText("downloadPanel.logInHint")).not.toBeNull();
+  });
+
+  test("Renders the log-in button", () => {
+    expect(container.getByText("downloadPanel.logInButton")).not.toBeNull();
+  });
+
+  test("Renders the sign-up button", () => {
+    expect(container.getByText("downloadPanel.signUpButton")).not.toBeNull();
+  });
+
+  test("Renders the donwload hint", () => {
+    expect(container.getByText("downloadPanel.downloadHint")).not.toBeNull();
+  });
+
+  test("Renders the download button", () => {
+    expect(container.getByText("downloadPanel.downloadButton")).not.toBeNull();
+  });
+
+  test("The download button initiates a download", async () => {
```

Comment:
The `DonwloadButton` has its own tests but felt that we should at least check it's doing something here given the issues we've had with the `DesignSystemButtons`